starlite/cache/memcached_cache_backend.py

```diff
@@ -0,0 +1,96 @@
+import pickle
+from typing import Any, Optional
+
+from pydantic import BaseModel
+
+from starlite.exceptions import MissingDependencyException
+
+try:
+    from aiomcache import Client
+except ImportError as e:
+    raise MissingDependencyException(
+        "To use starlite.memcached_cahce_backend, install starlite with 'memcached' extra, e.g. `pip install starlite[memcached]`"
+    ) from e
+
+
+from starlite.cache.base import CacheBackendProtocol
+
+
+class MemcachedCacheBackendConfig(BaseModel):
+    """Memcached cache backend configuration."""
+
+    host: str
+    """memcached host"""
+    port: Optional[int] = None
+    """memcached port (optional, defaults to 11211)"""
+    pool_size: Optional[int] = None
+    """Maximum number of memcached open connections (optional, defaults to 2)"""
+    pool_minsize: Optional[int] = None
+    """memcached minimum pool size (optional, by default set to `pool_size`)"""
+
+
+class MemcachedCacheBackend(CacheBackendProtocol):
+    def __init__(self, config: MemcachedCacheBackendConfig):
+        """This class offers a cache backend based on memcached.
+
+        Args:
+            config: required configuration to connect to memcached.
+        """
+        self._config = config
+        self._client: Client = None  # pyright: ignore
+
+    @property
+    def _memcached_client(self) -> Client:
+        if not self._client:
+            self._client = Client(**self._config.dict(exclude_unset=True))
+
+        return self._client
+
+    async def get(self, key: str) -> Any:  # pylint: disable=invalid-overridden-method
+        """Retrieves a value from cache corresponding to the given key.
+
+        Args:
+            key: name of cached value.
+
+        Returns:
+            Cached value if existing else `None`.
+        """
+
+        value = await self._memcached_client.get(key=key.encode())  # pyright: ignore
+        return pickle.loads(value)  # nosec
+
+    async def set(self, key: str, value: Any, expiration: int) -> None:  # pylint: disable=invalid-overridden-method
+        """Set sa value in cache for a given key for a duration determined by
+        expiration.
+
+        Args:
+            key: key to cache `value` under.
+            value: the value to be cached.
+            expiration: expiration of cached value in seconds.
+
+        Notes:
+            - expiration is in seconds.
+            - return value is not used by Starlite internally.
+
+        Returns:
+            None
+        """
+
+        await self._memcached_client.set(key.encode(), pickle.dumps(value), exptime=expiration)
```

Comment:
`pickle` is needed here because `aiomcache.Client.set()` requires both the key and value to be of type `bytes`. Maybe we should consider exposing the serialization method as part of `MemcachedCacheBackendConfig` and default it to `pickle`?

Reply:
sure

Reply:
Done in https://github.com/starlite-api/starlite/pull/576/commits/2de4c81122f95a6a6e6d32a41bf91231684da008